CMakeLists.txt

```diff
@@ -18,6 +18,20 @@ option(DOWNLOAD_STB_IMAGE "Download stb_image for image loading." ON)
 option(BUILD_GO_SHLIB "Build Go shared library." OFF)
 option(BUILD_DOCS "Build doxygen documentation (if doxygen is available)." ON)
 
+if(CMAKE_CROSSCOMPILING)
+  set(BOARD_NAME "" CACHE INTERNAL "Enter the board name to compile mlpack for.")
+  set(GFORTRAN_LIBRARY "" CACHE STRING "Path for gfortran library.")
+  set(PTHREAD_LIBRARY "" CACHE STRING "Path for Pthread library")
+  set(GCC_VERSION ""  CACHE STRING "")
```

Comment:
Is this ever used anywhere?  I think if GCC is being used, CMake actually has variables to get the version available anyway.

Reply:
Frankly, this variable is only for Ubuntu users that are usually on docker container who might use the default toolchain instead of buildroot, therefore, in this particular case, `aarch64-linux-gnu-gcc` might not exist, but instead it is often named as `aarch64-linux-gnu-gcc-5` with no dynamic link toward  `aarch64-linux-gnu-gcc`. In this case, the user might not be able at all to find the cross compiler, therefore I have added this variable, so the user can use it in this particular case

Reply:
Shouldn't they just override `CMAKE_CXX_COMPILER` then?  (Perhaps by modifying the toolchain file?)

Reply:
Of course, my initial thought was to reduce the user need to modify the toolchain file, but since it is a very rare case it will be more logical to remove the variable, and add mention this in documentation